### PR TITLE
move SymGrad and RCall kernels to functional_ops

### PR DESCRIPTION
**Pull request by gdh1995**
This moving of SymbolicGradient and RemoteCall is because
  their OPs are defined in ops/functional_ops.h .

Benefits:
if a C++ developer only imports ops/function_ops.cc and kernels/function_ops.cc,
  without 2 functional_ops.cc files,
then there won't be runtime errors reporing the OPs are not found.

Other influence:
This commit removes the 2 kernels from //tensorflow/core/kernels:android_core_ops,
  which is safe since the two OPs are not defined yet.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by alextp — DISMISSED** [review]

File: tensorflow/compiler/jit/xla_device_ops.h [hunk 1]
```diff
@@ -28,6 +28,7 @@ limitations under the License.
 #include "tensorflow/core/kernels/data/prefetch_dataset_op.h"
 #include "tensorflow/core/kernels/fifo_queue.h"
 #include "tensorflow/core/kernels/function_ops.h"
+#include "tensorflow/core/kernels/functional_ops.h"
```
Comment (yifeif) [inline tensorflow/compiler/jit/xla_device_ops.h]: Could you also update BUILD file?
`./tensorflow/compiler/jit/xla_device_ops.h:31:10: fatal error: 'tensorflow/core/kernels/functional_ops.h' file not found`

**Comment by yifeif** [comment]
Thanks for reviewing @alextp!

**Review by alextp — DISMISSED** [review]

**Review by alextp — APPROVED** [review]

File: tensorflow/core/kernels/functional_ops.cc [hunk 2]
```diff
@@ -15,24 +15,127 @@ limitations under the License.
...
+  AttrValueMap attr_values = func_.attr();
+  FunctionLibraryRuntime::InstantiateOptions instantiate_opts;
+  instantiate_opts.target = target_device;
+
+  FunctionTarget function_target = {target_device, lib};
+
+  FunctionLibraryRuntime::Handle handle;
+  {
+    mutex_lock l(mu_);
+    auto cached_entry = handle_cache_.find(function_target);
```
Comment (shahzadlone) [inline tensorflow/core/kernels/functional_ops.cc]: Could this be ```const```?

Comment (shahzadlone) [reply tensorflow/core/kernels/functional_ops.cc]: or does ```->second``` modify it?

File: tensorflow/core/kernels/functional_ops.cc [hunk 3]
```diff
@@ -15,24 +15,127 @@ limitations under the License.
...
+      arg_alloc_attrs.set_on_host(true);
+    }
+    opts.args_alloc_attrs.push_back(arg_alloc_attrs);
+  }
+  for (const auto& dtype : output_dtypes_) {
+    AllocatorAttributes ret_alloc_attrs;
+    if (DataTypeAlwaysOnHost(dtype)) {
+      ret_alloc_attrs.set_on_host(true);
+    }
+    opts.rets_alloc_attrs.push_back(ret_alloc_attrs);
```
Comment (shahzadlone) [inline tensorflow/core/kernels/functional_ops.cc]: could this be reserved? ```opts.rets_alloc_attrs```

**Comment by rthadur** [comment]
@gdh1995 gentle ping to resolve conflicts

**Comment by gdh1995** [comment]
Sorry but recent days I have no enough time to complete this PR. So let me close it.

**Pull request closed without merging**